HW3/P3/sum.cl

```diff
@@ -48,13 +64,14 @@ __kernel void sum_blocked(__global float* x,
     // 
     // Be careful that each thread stays in bounds, both relative to
     // size of x (i.e., N), and the range it's assigned to sum.
-    for (;;) { // YOUR CODE HERE
-        ; // YOUR CODE HERE
+    for (i=global_id*k; i <= (k*(global_id+1)-1); i++) { // YOUR CODE HERE
+        sum=sum+x[i]; 
```

Comment:
You should verify that you access only valid elements from x (i.e., that i < N).